test/test_metric/test_metrics.py

```diff
@@ -684,6 +684,56 @@ def test_calculate_losses():
         assert pytest.approx(expected_score) == score_dict[expected_metric]
         assert pytest.approx(1 - expected_score) == loss_dict[expected_metric]
 
+    # Test no metric
+    with pytest.raises(
+        ValueError, match="Number of metrics to compute must be greater than zero."
+    ):
+        calculate_scores(
+            solution=y_true,
+            prediction=y_pred,
+            task_type=BINARY_CLASSIFICATION,
+            metrics=[],
+        )
+
+    with pytest.raises(
+        ValueError, match="Number of metrics to compute must be greater than zero."
+    ):
+        calculate_scores(
+            solution=y_true,
+            prediction=y_pred,
+            task_type=BINARY_CLASSIFICATION,
+            metrics=[],
+            scoring_functions=[
+                autosklearn.metrics.accuracy,
+                autosklearn.metrics.balanced_accuracy,
+            ],
+        )
+
+    # Test the same metric twice
+    accuracy_fixture = {"accuracy": pytest.approx(0.9)}
+    assert accuracy_fixture == calculate_scores(
+        solution=y_true,
+        prediction=y_pred,
+        task_type=BINARY_CLASSIFICATION,
+        metrics=[autosklearn.metrics.accuracy, autosklearn.metrics.accuracy],
```

Comment:
[Blocking]
Using the same metric twice seems like it should raise an error to me. Given that the behavior is no error raised, the output is good though

Reply:
Do you mean we should allow the same metric to be present in both `metrics` and `scoring_functions`, but not the same metric twice in one of them?

Reply:
I guess so, I didn't really think about the same one twice, once in `metrics` and once in `scoring_functions`.

What you stated seems reasonable to me but not a cause to block the PR if it's difficult. I can't imagine a scenario where you would want the same metric, i.e. `acc` twice in `metrics` but I could imagine a scenario where you have `acc` in `metrics` and `acc` in `scoring_functions`.

The scenario where I see this being used is purely to make getting out scores for autosklearn to be done in one place, i.e. you specify `acc` and `metric_z` for the optimization with `metrics` and you specify `acc`, `balanced_acc` and `f1_score` for the `scoring_functions` when you later want to evaluate the autosklearn run.